spec/controllers/instructeurs/dossiers_controller_spec.rb

```diff
@@ -1342,4 +1342,27 @@
 
     it { expect(subject).to have_http_status(:ok) }
   end
+
+  describe '#pieces_jointes' do
+    let(:procedure) { create(:procedure, :published, types_de_champ_public: [{ type: :piece_justificative }], instructeurs:) }
+    let(:dossier) { create(:dossier, :en_construction, :with_populated_champs, procedure: procedure) }
+
+    before do
+      dossier.champs.first.piece_justificative_file.attach(
+        io: StringIO.new("image file"),
+        filename: "image.jpeg",
+        content_type: "image/jpeg",
+        # we don't want to run virus scanner on this file
+        metadata: { virus_scan_result: ActiveStorage::VirusScanner::SAFE }
+      )
+      get :pieces_jointes, params: {
+        procedure_id: procedure.id,
+        dossier_id: dossier.id
+      }
+    end
+
+    it { expect(response.body).to include('Télécharger le fichier toto.txt') }
+    it { expect(response.body).to include('Télécharger le fichier image.jpeg') }
+    it { expect(response.body).to include('Visualiser') }
```

Comment:
Pour les perfs de tests il vaut mieux regrouper tous les expect des mêmes conditions dans un même it, pour éviter de refaire 3x tout le setup (procedure/dossier/user/ attachment etc… qui prend des plombes).

```suggestion
    it "renders links" do
      expect(response.body).to include('Télécharger le fichier toto.txt')
      expect(response.body).to include('Télécharger le fichier image.jpeg')
      expect(response.body).to include('Visualiser')
    end
```